backend/src/apiserver/server/pipeline_server.go

```diff
@@ -191,11 +191,14 @@ func (s *PipelineServer) CreatePipelineV1(ctx context.Context, request *apiv1bet
 	}
 
 	// A new pipeline has been created, but pipeline version creation failed after validation
+	// This block should not be executed when an existing pipeline is found
 	if perr == nil {
-		return toApiPipelineV1(createdPipeline, &model.PipelineVersion{}), nil
+		if cleanupErr := s.deletePipeline(ctx, createdPipeline.UUID); cleanupErr != nil {
```

Comment:
While the reverting logic is what have discussed and it should be fine in probably most if not all cases, a better solution would be make creating pipeline and pipeline version a DB transaction, so that no reverting would ever be needed. WDYT? 
While it should be technically feasible, I'm not sure the implementation cost. So I'm fine to have a reverting logic for now if the transaction logic requires too much effort to change.

Reply:
Good point thank you.

Let me implement a single-transaction call for creating v1 pipelines.